xdsl/dialects/riscv.py

```diff
@@ -707,6 +707,19 @@ def __init__(
     def assembly_line_args(self) -> tuple[AssemblyInstructionArg, ...]:
         return self.rd, self.rs1, self.immediate
 
+    @classmethod
+    def custom_parse_attributes(cls, parser: Parser) -> Mapping[str, Attribute]:
+        attributes = dict[str, Attribute]()
+        attributes["immediate"] = _parse_immediate_value(
+            parser, IntegerType(12, Signedness.SIGNED)
```

Comment:
what about the child class `RdRsImmShiftOperation` which uses 5 bits immediate ?

Reply:
Address in #1416